cep-xx.md

```diff
@@ -0,0 +1,155 @@
+As a comparison, this would be like searching for a decimal to represent the lower bound of the
+interval ``>π`` (the mathematical constant).
+Hence, the ``*`` character, conveniently not used in practice in versions, was used to represent
+this "version" that does not exist but that is useful for representing intervals
+(``>1.1*,<1.2*`` in the above example).
+We refer to this "imaginary" version as the _``1.1`` infimum_ in the rest of this proposal.
+
+But that special version should always have stayed an implementation detail and never leak into
+the set of valid versions, as allowing so breaks it purpose when considering the spec ``1.1*.0*``.
+The special character ``*`` should also have been limited to a single appearance per version,
+otherwise we can repeat the infinitly decreasing sequence ``1.1* > 1.1*.0* > 1.1*.0*.0* > ...``
+of versions in that start with ``1.1``.
+
+
+## Rationale
+We searched through the ``repodata.json`` of all subdirs of the ``anaconda``, ``conda-forge``,
+and ``bioconda`` channels and found that:
+ - No ``"version"`` field of any package contains a ``*``
+ - Occurences of ``>1.1*``, ``>1.1.*``, ``>=1.1*``, ``>=1.1.*``
+ - Few occurences of ``<1.1*`` (on bioconda), ``<1.1.*``, no occurence of ``<=1.1*`` and ``<=1.1.*``
+ - Single occurence of ``==2.6*`` (dep on Python bioconda neve matching) and no occurence of ``=1.1*``
+
+
+## Specification
+### [V1] Version
+Given that no version currently contains the ``*`` character, we reommend to simply forbid ``*``
+inside a version.
+ - The public API of ``Version`` must fail to create/parse a version from string contaning ``*``.
+ - ``Version`` must not serialize to any string containing a ``*``.
+ - ``repodata.json`` and the like must not contain a ``*`` in any of the ``"version"`` fields.
+
+Note that ``*`` can still be used internally to represent these special infimum "versions".
+
+### VersionSpec
+#### [VS1] Equality and startswith
+For equality (``==`` and no prefix operators) and starts with (``=`` operator) we advocate to always
+interpret ``*`` as a glob expression.
+From the motivation section, this would change the behaviour of ``=1.7*`` and ``==1.7*``.
+Since no version where found containing a ``*``, the only occurence of such version spec are
+most likely a mistake.
+
+We further argue to make a difference between ``.*`` and ``*`` suffixes to make their meaning
+closer to glob expresions.
+- ``1.7*`` should match version starting with ``1.7``, that is for instance _including_ ``1.7alpha``;
+- ``1.7.*`` should match anything starting with ``1.7.``, that is _excluding_ ``1.7alpha``.
+
+Currently, in ``conda``, only ``=1.7`` is matching ``1.7alpha``.
+
+#### [VS2] Relational comparison operators ``>``, ``>=``, ``<``, ``<=``
+This is the most delicate part to handle since ``>1.7*`` is frequently use to mean greater than the
+``1.7`` infimum (could also be writen as ``=1.7|>=1.7``, that is, mathching some versions smaller
+than ``1.7.0``, such as ``1.7alpha``).
```

Comment:
Existing cases here are often caused by someone accidentally using `=>1.7*` in a recipe instead of `>=1.7*`. I vote we should interpret it as what is most likely intended here (e.g. `>=1.7*`).